tests/UnitTests/PhasorDynamics/GenrouTests.hpp

```diff
@@ -101,6 +102,113 @@ namespace GridKit
         return success.report(__func__);
       }
 
+      // A test to verify that the hard coded answers match those given by the residual functions
+      // Hard code parameters, differential, and algebraic terms
+      TestOutcome hard_coded_residual()
+      {
+        TestStatus success = true;
+
+        // GenRou generator parameters
+        real_type p0{1};
+        real_type q0{.05013};
+        real_type H{3};
+        real_type D{.5};
+        real_type Ra{.1};
+        real_type Tdop{7};
+        real_type Tdopp{.04};
+        real_type Tqopp{.05};
+        real_type Tqop{.75};
+        real_type Xd{2.1};
+        real_type Xdp{.2};
+        real_type Xdpp{.5};
+        real_type Xq{.18};
+        real_type Xqp{.3};
+        real_type Xqpp{.5};
+        real_type Xl{.5};
+        real_type S10{.1};
+        real_type S12{.2};
+
+        ScalarT Vr1{1.0}; ///< Bus real voltage
+        ScalarT Vi1{0};   ///< Bus imaginary voltage
+
+        PhasorDynamics::Bus<ScalarT, IdxT>    bus(Vr1, Vi1);
+        PhasorDynamics::Genrou<ScalarT, IdxT> gen(&bus, 1, p0, q0, H, D, Ra, Tdop, Tdopp, Tqopp, Tqop, Xd, Xdp, Xdpp, Xq, Xqp, Xqpp, Xl, S10, S12);
+
+        // Answer key is available only in double precision.
+        // Therefore, only double precision tests are done at this time.
+        const std::vector<double> res_answer = {
```

Comment:
The solution vector elements are of type `ScalarT`. In the instantiation of tests, we instantiate only the class with template parameter `ScalarT==double`
```suggestion
        const std::vector<ScalarT> res_answer = {
```